ci/docker/runtime_functions.sh

```diff
@@ -66,37 +66,28 @@ build_ccache_wrappers() {
     # But in the beginning, we'll make this opt-in. In future, loads of processes like
     # the scala make step or numpy compilation and other pip package generations
     # could be heavily sped up by using ccache as well.
-    mkdir /tmp/ccache-redirects
+    mkdir -p /tmp/ccache-redirects
     export PATH=/tmp/ccache-redirects:$PATH
-    ln -s ccache /tmp/ccache-redirects/gcc
-    ln -s ccache /tmp/ccache-redirects/gcc-8
-    ln -s ccache /tmp/ccache-redirects/g++
-    ln -s ccache /tmp/ccache-redirects/g++-8
-    ln -s ccache /tmp/ccache-redirects/nvcc
-    ln -s ccache /tmp/ccache-redirects/clang++-3.9
-    ln -s ccache /tmp/ccache-redirects/clang-3.9
-    ln -s ccache /tmp/ccache-redirects/clang++-5.0
-    ln -s ccache /tmp/ccache-redirects/clang-5.0
-    ln -s ccache /tmp/ccache-redirects/clang++-6.0
-    ln -s ccache /tmp/ccache-redirects/clang-6.0
-    ln -s ccache /usr/local/bin/gcc
-    ln -s ccache /usr/local/bin/gcc-8
-    ln -s ccache /usr/local/bin/g++
-    ln -s ccache /usr/local/bin/g++-8
-    ln -s ccache /usr/local/bin/nvcc
-    ln -s ccache /usr/local/bin/clang++-3.9
-    ln -s ccache /usr/local/bin/clang-3.9
-    ln -s ccache /usr/local/bin/clang++-5.0
-    ln -s ccache /usr/local/bin/clang-5.0
-    ln -s ccache /usr/local/bin/clang++-6.0
-    ln -s ccache /usr/local/bin/clang-6.0
```

Comment:
Please also restore the links in /usr/local/bin/

Reply:
no need, also they fail with permission denied when executing locally or without root privileges. 

No need because they are added to the path.

Reply:
Are you aware that the links you created previously were dangling?  (basically didn't work)

Reply:
Why do you think the links to /usr/local/bin should be restored? Could you explain?

Reply:
We spoke about this already a dozen times, so I will only link the official guide: https://ccache.dev/manual/latest.html#_run_modes

Back when I developed this, I tested it only by relaying on the PATH and it didn't work. Thus, I added the /usr/local/bin method. 

The links did work - why shouldn't they?

Reply:
It didn't work because your links were broken. So with my changes it works in /tmp/ccache-redirects . there's no need to touch /usr/local/bin this is not allowed for users, requires sudo privileges, it's not necessary, etc. Please elaborate why you don't accept my fix, which uses the ccache wrappers in the /tmp folder. The guide says "something like this" you are taking the guide too seriously, having in any other folder which is then injected first in the path is enough.

Reply:
Can you elaborate on why are you requesting changes when my changes actually fix the problem and make ccache work? or do you think there is something that is not working with my changes? Is not clear from your comments.

Reply:
You are not answering my comments, can you get into the container and see that this PR is fixing your broken links? That's the reason your approach didn't work, and you still left the /tmp/ccache-redirect broken links. I fixed this, thus no need to write in /usr/local/bin which prevents the build functions from being used locally or without root permissions.